### PR TITLE
TLS Private Key Data Source

### PR DESCRIPTION
**Pull request by bengesoff**
Heavily inspired by Will's TLS Configuration data source. 😉

This won't work correctly yet for two reasons:
- It depends on a PR in `go-fastly` (https://github.com/opencredo/go-fastly/pull/26)
- It depends on the associated private key resource PR for the tests to work, and will need rebasing (https://github.com/opencredo/terraform-provider-fastly/pull/70)

### CODE REVIEW

**Comment by bengesoff** [comment]
Ps: docs coming soon

File: go.mod [hunk 1]
```diff
@@ -2,6 +2,8 @@ module github.com/fastly/terraform-provider-fastly
 
 go 1.14
 
+replace github.com/fastly/go-fastly/v2 v2.1.0 => github.com/opencredo/go-fastly/v2 v2.0.0-20210120105619-a13767a18fb8
```
Comment (bengesoff) [inline go.mod:5]: Temporary commit reference here, will need to update when go-fastly branch is set up correctly

File: fastly/data_source_tls_private_key_test.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	name := fmt.Sprintf("tf-test-%s", acctest.RandString(10))
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                  func() { testAccPreCheck(t) },
+		Providers:                 testAccProviders,
+		PreventPostDestroyRefresh: true, // Testing data source so state refresh will fail after destroy
+		Steps: []resource.TestStep{
+			{
+				// Just the resource, no data sources, so no checks needed
+				Config: testAccFastlyDataSourceTLSPrivateKeyConfigResources(key, name),
```
Comment (wjam) [inline fastly/data_source_tls_private_key_test.go:27]: Why not put the resource and data source HCL into the same text block like https://github.com/hashicorp/terraform-provider-aws/blob/master/aws/data_source_aws_s3_bucket_test.go#L56?

File: fastly/data_source_tls_private_key_test.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+		t.Fatalf("Failed to generate private key: %v", err)
+	}
+	key = strings.ReplaceAll(key, "\n", `\n`)
+
+	name := fmt.Sprintf("tf-test-%s", acctest.RandString(10))
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                  func() { testAccPreCheck(t) },
+		Providers:                 testAccProviders,
+		PreventPostDestroyRefresh: true, // Testing data source so state refresh will fail after destroy
```
Comment (wjam) [inline fastly/data_source_tls_private_key_test.go:23]: Not sure why this would be needed?

File: fastly/data_source_tls_private_key_test.go (hunk 2 above)
Comment (bengesoff) [reply fastly/data_source_tls_private_key_test.go:27]: I thought I was having some trouble with race conditions between the resource and data source, but I think that must have been the issue I was having with the `PageNumber` not working properly as I was discussing on your TLS Configuration PR 🤔  good spot

File: fastly/data_source_tls_private_key_test.go (hunk 3 above)
Comment (bengesoff) [reply fastly/data_source_tls_private_key_test.go:23]: I was having some problems with the state not being empty after destroying and I saw this being used in an AWS resource test to resolve it, but you seem to be right in that it isn't needed

**Comment by bengesoff** [comment]
Website test failing, will rebase onto @wjam's branch with the fix when that's merged

**Pull request merged**